### PR TITLE
replacement for scripts/ansible & and exit codes for runansible

### PR DESCRIPTION
**Pull request by jvonau**

### CODE REVIEW

**Comment by jvonau** [comment]
Seen snowballing in #313 runansible exited with 'OS not supported' but the master script continued. With this PR in place the "web load scripts" can go back to using scripts/ansible to install ansible and the logic upkeep is back in one place under git control.

**Comment by jvonau** [comment]
um.. this for 6.4 so the "install from scratch" works as what the [docs](https://github.com/iiab/iiab/wiki/IIAB-Installation1) have across all distros. or start making load scripts for CentOS

**Comment by jvonau** [comment]
version 2.4.0 installed on raspbian-9 as a test.. 
[byhand-rasp-9.txt](https://github.com/iiab/iiab/files/1338393/byhand-rasp-9.txt)
and updated this PR to reflect what is need to enable that across the debian family.
[2.4.0-log.txt](https://github.com/iiab/iiab/files/1338394/2.4.0-log.txt)

**Comment by tim-moody** [comment]
just cloned  https://github.com/jvonau/iiab -b 2.3.1-patch on ubuntu 16.04

runansible:

TASK [1-prep : Checking iiab_domain_name] ****************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": "cat /etc/sysconfig/iiab_domain_name", "delta": "0:00:00.003372", "end": "2017-09-27 17:20:33.975591", "failed": true, "msg": "non-zero return code", "rc": 1, "start": "2017-09-27 17:20:33.972219", "stderr": "cat: /etc/sysconfig/iiab_domain_name: No such file or directory", "stderr_lines": ["cat: /etc/sysconfig/iiab_domain_name: No such file or directory"], "stdout": "", "stdout_lines": []}

**Comment by jvonau** [comment]
log please..

**Comment by tim-moody** [comment]
http://paste.ubuntu.com/25629975/

**Comment by tim-moody** [comment]
root@ubuntu-base:/opt/iiab/iiab# apt list ansible -a
Listing... Done
ansible/xenial,xenial,now 2.4.0.0-1ppa~xenial all [installed]
ansible/xenial-backports,xenial-backports 2.1.1.0-1~ubuntu16.04.1 all
ansible/xenial-updates,xenial-updates 2.0.0.2-2ubuntu1 all
ansible/xenial,xenial 2.0.0.2-2 all

**Comment by tim-moody** [comment]
I would prefer not to have runansible install ansible

**Comment by jvonau** [comment]
> 2017-09-27 17:59:06,249 p=8795 u=root |  PLAY [all] **********************************************************************************************************************************
2017-09-27 17:59:07,024 p=8795 u=root |  TASK [Gathering Facts] *******************************************************************************************************************************
2017-09-27 17:59:08,690 p=8795 u=root |  ok: [127.0.0.1]
2017-09-27 17:59:08,703 p=8795 u=root | TASK [1-prep : Determine if runansible was run] *********************************************************************************************************************************2017-09-27 17:59:09,000 p=8795 u=root | ok: [127.0.0.1]
2017-09-27 17:59:09,014 p=8795 u=root |  TASK [1-prep : Setting first run flag] **********************************************************************************************************************************
2017-09-27 17:59:09,039 p=8795 u=root |  skipping: [127.0.0.1]
2017-09-27 17:59:09,052 p=8795 u=root |  TASK [1-prep : Create iiab config file] **********************************************************************************************************************************
2017-09-27 17:59:09,075 p=8795 u=root |  skipping: [127.0.0.1]

Think you have a /etc/iiab/config_vars.yml on the box.. otherwise the next stanza would have not be skipped.

>  "ansible_program": "/usr/local/bin/ansible"
                "ansble_version": "2.4.0.0", 
                "iiab_branch": "2.3.1-patch",
 
That would imply that 2.2.0.0 was pip installed in the past and the upgrade process worked as intended from scripts/ansible.

**Comment by jvonau** [comment]
What the output should be.
> 2017-09-27 12:32:33,875 p=11563 u=root |  PLAY [all] *********************************************************************
2017-09-27 12:32:38,752 p=11563 u=root |  TASK [setup] *******************************************************************
2017-09-27 12:32:44,297 p=11563 u=root |  ok: [127.0.0.1]
2017-09-27 12:32:44,419 p=11563 u=root |  TASK [1-prep : Determine if runansible was run] ********************************
2017-09-27 12:32:45,671 p=11563 u=root |  ok: [127.0.0.1]
2017-09-27 12:32:45,762 p=11563 u=root |  TASK [1-prep : Setting first run flag] *****************************************
2017-09-27 12:32:46,000 p=11563 u=root |  ok: [127.0.0.1]
2017-09-27 12:32:46,090 p=11563 u=root |  TASK [1-prep : Create iiab config file] ****************************************
2017-09-27 12:32:47,644 p=11563 u=root |  changed: [127.0.0.1]
2017-09-27 12:32:47,733 p=11563 u=root |  TASK [1-prep : Add location section to config

**Comment by tim-moody** [comment]
if runansible puts /etc/iiab/config_vars.yml  and fails then it will be there for the next run

**Comment by holta** [comment]
"./runansible" (v2.4.0.0) smoke tests successfully & 13 of my 14 server apps pass their basic functional tests(*) on Raspbian Stretch Lite after running this 1-line installer:

http://download.iiab.io/6.4/rpi/load-ansible2.4.0-vpn.txt

Which calls:
```
git clone https://github.com/jvonau/iiab -b 2.3.1-patch
```
(*) functional testing of all the 14 basic server apps listed in [Medium-sized local_vars.yml](http://wiki.laptop.org/go/IIAB/local_vars.yml) as summarized at http://FAQ.IIAB.IO Question 3 ("What can I do with E-books and Internet-in-a-Box?").  The 14th test I ran was "Install Configured Options" which worked within http://box/admin e.g. to install http://box/elgg (2.3.4)

_EXCEPTION: http://box/usb fails to show [teacher content](http://wiki.laptop.org/go/IIAB/FAQ#Can_teachers_display_their_own_content.3F) as has been happening to all Raspbian systems in the past ~24h or so.  Definitely a regression relating to #195 (usbmount) we need to address!_

**Comment by tim-moody** [comment]
good to hear it runs on stretch. please note above I said I was on ubuntu.

**Comment by jvonau** [comment]
Um, that VM was not pristine, as in never had any iiab work applied to the test subject. I found in the log the path this VM was using was /usr/local/bin/ansible, which is the pip path. If this was a pristine test subject scripts/ansible would of taken a different path, enabled ansibles' ppa installing via .debs and as you already know the resulting path would be /usr/bin/ansible. Thank you never the less as this confirms that pip upgrades from 2.2.0.0 -> 2.4.0.0 work seamlessly. If /etc/iiab was renamed before the test, that would of been closer to pulling off faking 'pristine'.

**Comment by jvonau** [comment]
> if runansible puts /etc/iiab/config_vars.yml and fails then it will be there for the next run

see https://github.com/iiab/iiab/commit/e57ff51ca8a280c5050ba71ba8801bba1b792a40

**Comment by tim-moody** [comment]
It was pristine when I started the test. When the test failed, as reported above, I tried to install 6.3 and then went back to this patch. Unfortunately the log is from the second attempt.

**Comment by tim-moody** [comment]
actually, I tried so many things it may have had current master run before 

 cloned https://github.com/jvonau/iiab -b 2.3.1-patch on ubuntu 16.04

**Comment by tim-moody** [comment]
retesting

**Comment by jvonau** [comment]
I'd just rename/remove /etc/iiab and runansible, the noted stoppage should be gone.

**Comment by tim-moody** [comment]
fresh vm runansible never run
old scripts/ansible has been run

**Comment by holta** [comment]
Raspbian-side Clarif &mdash; Elgg 2.3.4 is the only Functional Tests I've found that's failed on the RPi built with Ansible 2.4.0.0 (likely an actual regression).  While Elgg 2.3.4's playbook installed onto 192.168.0.132 = 10.8.0.46 (according to "Install Configured Options" and "Display Job Status") but the result at http://box/elgg does not load in browser however.  This is what appears in browser:
```
Fatal error in exception handler. Check log for Exception at time 1506558199
```
On reboot, it shows:
```
Fatal error in exception handler. Check log for Exception at time 1506558624
```

Elgg prior refs: #217 #306 #307


TOTAL ASIDE: Running "systemctl" shows these 3 lines in red all RPi systems (built on Ansible 2.2.1 or 2.4.0) for now, which may or may not be a concern:
```
  isc-dhcp-server.service      loaded failed failed    LSB: DHCP server
  mongodb.service              loaded failed failed    High-performance, schema-free document-oriented database
  sugarizer.service            loaded failed failed    sugarizer.service
```

**Comment by tim-moody** [comment]
got past previous error

**Comment by tim-moody** [comment]
ran to completion with a medium set of services

**Comment by jvonau** [comment]
@holta The three services I'm aware of and have fixes in branches to address those. Put it on the table for tomorrow.

File: scripts/ansible [hunk 1]
```diff
@@ -1,106 +1,135 @@
...
-    fi
-    # If python-pip install failed and setuptools exists, try that
-    if [ -z "$(which pip)" -a -z "$(which easy_install)" ]; then
-      apt-get -y install python-setuptools
-      easy_install pip
-    elif [ -z "$(which pip)" -a -n "$(which easy_install)" ]; then
-      easy_install pip
+#  handle 2.2.1 2.2.0.0 <-> 2.4.0
+if [ $FAMILY = "debian" ]; then
+    if [ $GOOD_VER -lt 1 ]; then
```
Comment (holta) [inline scripts/ansible]: The error "line 110: [: -lt: unary operator expected" below implies that $GOOD_VER was never set (or at least has no value!) here on Raspbian anyway &mdash; when running [download.iiab.io/6.4/rpi/load-ansible2.4.0-vpn.txt](http://download.iiab.io/6.4/rpi/load-ansible2.4.0-vpn.txt) :

```
Setting up python-paramiko (2.0.0-1) ...
Setting up ansible (2.4.0.0-1ppa~xenial) ...
Processing triggers for libc-bin (2.24-11+deb9u1) ...
./scripts/ansible: line 110: [: -lt: unary operator expected
Current ansible version installed is 2.4.0.0
Running local playbooks!

PLAY [all] *********************************************************************
```

Comment (jvonau) [reply scripts/ansible]: Changed the logic a little please review

**Pull request merged**